docs/understand/cmake_packages.rst

```diff
@@ -1,139 +1,183 @@
+dependencies are not found in standard system locations, one either has to
+instruct CMake to search for package config files in additional folders using
+the ``CMAKE_PREFIX_PATH`` variable (a semi-colon separated list of filesystem
+paths), or using ``<PackageName>_ROOT`` variable on a project-specific basis.
+
+There are nearly a dozen ways to set these variables. One may be more convenient
+over the other depending on your workflow. Conceptually the simplest is adding
+it to your CMake configuration command on the command-line via
+``-D CMAKE_PREFIX_PATH=....`` . AMD packaged ROCm installs can typically be
+added to the config file search paths such as:
+
+-  Windows: ``-D CMAKE_PREFIX_PATH=${env:HIP_PATH}``
+
+-  Linux: ``-D CMAKE_PREFIX_PATH=/opt/rocm``
 
-There are two ways to set this variable:
+ROCm provides the respective *config-file* packages, and this enables
+``find_package`` to be used directly. ROCm does not require any Find module as
+the *config-file* packages are shipped with the upstream projects.
 
--  Pass the flag when configuring with ``-DCMAKE_PREFIX_PATH=....`` This
-   approach is preferred when users install the components in custom locations.
+For a complete guide on where and how ROCm may be installed on a system, refer
+to the installation guides in these docs (`Linux <../deploy/linux/index.html>`_).
 
--  Append the variable in the CMakeLists.txt file. This is useful if the
-   dependencies are found in a common location. For example, when the binaries
-   provided on `repo.radeon.com <http://repo.radeon.com>`_ are installed to
-   */opt/rocm*, you can add the following line to a CMakeLists.txt file
+Using HIP in CMake
+==================
 
-   ::
+ROCm componenents providing a C/C++ interface support being consumed using any
+C/C++ toolchain that CMake knows how to drive. ROCm also supports CMake's HIP
+language features, allowing users to program using the HIP single-source
+programming model. When a program (or translation-unit) uses the HIP API without
+compiling any GPU device code, HIP can be treated in CMake as a simple C/C++
+library.
 
-    list (APPEND CMAKE_PREFIX_PATH /opt/rocm/hip /opt/rocm)
+Using the HIP single-source programming model
+---------------------------------------------
 
-Using HIP in CMake
---------------------
+Source code written in the HIP dialect of C++ typically uses the `.hip`
+extension. When the HIP CMake language is enabled, it will automatically
+associate such source files with the HIP toolchain being used.
 
-There are two ways to use HIP in CMake:
+::
 
--  Use the HIP API without compiling the GPU device code. As there is no GPU
-   code, any C or C++ compiler can be used. The ``find_package(hip)`` provides
-   the ``hip::host`` target to use HIP in this context
+    cmake_minimum_required(VERSION 3.21) # enable_language(HIP)
+    cmake_policy(VERSION 3.21.3...3.27)
+    project(MyProj LANGUAGES HIP)
+    add_executable(MyApp Main.hip)
```

Comment:
Considering users may have pre-existing cuda source files. Adding an example can be helpful.
```
add_library(foo foo.cu)
set_source_files_properties(foo.cu PROPERTIES LANGUAGE HIP)
```

Reply:
Added.